api/api.js

```diff
@@ -0,0 +1,22 @@
+export default{
+    async getData(){
+        const response = await fetch("https://rickandmortyapi.com/api/character");
+        return response.json();
+    },
+    async getDataById(id){
+        const response = await fetch(`https://rickandmortyapi.com/api/character/${id}`);
+        return response.json();
+    },
+    async getDataByName(name){
```

Comment:
No need to create separate functions, pass query params object in this function and generate the URL with query params